jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/transport/HttpReceiver.java

```diff
@@ -740,6 +755,15 @@ public void onDataAvailable()
             invokeDemandCallback(false);
         }
 
+        @Override
+        public InvocationType getInvocationType()
+        {
+            Runnable demandCallback = demandCallbackRef.get();
+            if (demandCallback == null)
```

Comment:
There should definitely be a comment explaining why we return the connection's invocation type when the demand callback is null, as that's quite a special trick for which I already forgot the details of.

Reply:
There is not much to say: if I don't have the demand callback, I cannot extract the invocation type from it, and therefore we fall back to the connection invocation type.